src/host.rs

```diff
@@ -70,11 +198,31 @@ impl Host {
             instructions,
             message,
         });
-        self.version += 1;
 
         self.notify.notify_one();
     }
 
+    pub(crate) fn embark_on(
```

Comment:
Embark, like getting on a ship. You're not yet at your destination - you're just on the ship. Maybe I picked this name because I was thinking along the line of ports.. Either way, I don't love the name, but can't think of anything better. Maybe enqueue?

Send was ambiguous in so many ways. Was the host sending something (no.. it was preparing to receive something). When the method returns is it (the message) then sent (no.. it's in a pipe somewhere).